go-controller/pkg/node/node_smartnic.go

```diff
@@ -0,0 +1,203 @@
+					retryPods.Store(pod.UID, true)
+					return
+				}
+				podInterfaceInfo, err := cni.PodAnnotation2PodInfo(pod.Annotations, true)
+				if err != nil {
+					retryPods.Store(pod.UID, true)
+					return
+				}
+				err = n.addRepPort(pod, vfRepName, podInterfaceInfo)
+				if err != nil {
+					klog.Infof("Failed to add rep port, %s. retrying", err)
+					retryPods.Store(pod.UID, true)
+				} else {
+					servedPods.Store(pod.UID, true)
+				}
+			} else {
+				// Handle unscheduled pods later in UpdateFunc
+				retryPods.Store(pod.UID, true)
+				return
+			}
+		},
+		UpdateFunc: func(old, newer interface{}) {
+			pod := newer.(*kapi.Pod)
+			klog.Infof("UpdateFunc for POD: %s/%s", pod.ObjectMeta.GetNamespace(), pod.ObjectMeta.GetName())
+			if !util.PodWantsNetwork(pod) || pod.Status.Phase == kapi.PodRunning {
+				retryPods.Delete(pod.UID)
+				return
+			}
+			_, retry := retryPods.Load(pod.UID)
+			if util.PodScheduled(pod) && retry {
+				if n.name != pod.Spec.NodeName {
+					retryPods.Delete(pod.UID)
+					return
+				}
+				vfRepName, err := n.getVfRepName(pod)
+				if err != nil {
+					klog.Infof("Failed to get rep name, %s. retrying", err)
+					retryPods.Store(pod.UID, true)
+					return
+				}
+				podInterfaceInfo, err := cni.PodAnnotation2PodInfo(pod.Annotations, true)
+				if err != nil {
+					retryPods.Store(pod.UID, true)
+					return
+				}
+				err = n.addRepPort(pod, vfRepName, podInterfaceInfo)
+				if err != nil {
+					klog.Infof("Failed to add rep port, %s. retrying", err)
+					retryPods.Store(pod.UID, true)
+				} else {
+					servedPods.Store(pod.UID, true)
+					retryPods.Delete(pod.UID)
+				}
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			pod := obj.(*kapi.Pod)
+			klog.Infof("DeleteFunc for POD: %s/%s", pod.ObjectMeta.GetNamespace(), pod.ObjectMeta.GetName())
+			if _, ok := servedPods.Load(pod.UID); !ok {
+				return
+			}
+			servedPods.Delete(pod.UID)
+			retryPods.Delete(pod.UID)
+			vfRepName, err := n.getVfRepName(pod)
+			if err != nil {
+				klog.Errorf("Failed to get VF Representor Name from Pod: %s. Representor port may have been deleted.", err)
+				return
+			}
+			err = n.delRepPort(vfRepName)
+			if err != nil {
+				klog.Infof("Failed to delete VF representor %s. %s", vfRepName, err)
+			}
+		},
+	}, nil)
+}
+
+// getVfRepName returns the VF's representor of the VF assigned to the pod
+func (n *OvnNode) getVfRepName(pod *kapi.Pod) (string, error) {
+	smartNicCD := util.SmartNICConnectionDetails{}
+	if err := smartNicCD.FromPodAnnotation(pod.Annotations); err != nil {
+		return "", fmt.Errorf("failed to get smart-nic annotation. %v", err)
+	}
+	return util.GetSriovnetOps().GetVfRepresentorSmartNIC(smartNicCD.PfId, smartNicCD.VfId)
+}
+
+// addRepPort adds the representor of the VF to the ovs bridge
+func (n *OvnNode) addRepPort(pod *kapi.Pod, vfRepName string, ifInfo *cni.PodInterfaceInfo) error {
+	klog.Infof("addRepPort: %s", vfRepName)
```

Comment:
lets not use the function name in the log message. Change it to something more obvious like "Adding VF representor %s for pod %s"

Reply:
will be addressed

Reply:
need to fix in this commit